pkg/resource/resource.go

```diff
@@ -632,8 +462,194 @@ func sendNodesMetrics(obj *unstructured.Unstructured, crName string) {
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" ||
+		kind == "CSIDriver" ||
+		kind == "Issuer" ||
+		kind == "CustomResourceDefinition" ||
+		kind == "Certificate" ||
+		kind == "SpecialResource" ||
+		kind == "OperatorGroup" ||
+		kind == "CertManager" ||
+		kind == "MutatingWebhookConfiguration" ||
+		kind == "ValidatingWebhookConfiguration" ||
+		kind == "Deployment" ||
+		kind == "ImagePolicy" {
+		return true
+	}
+	return false
+
+}
+
+func UpdateResourceVersion(req *unstructured.Unstructured, found *unstructured.Unstructured) error {
+
+	kind := found.GetKind()
+
+	if NeedsResourceVersionUpdate(kind) {
+		version, fnd, err := unstructured.NestedString(found.Object, "metadata", "resourceVersion")
+		if err != nil || !fnd {
+			return fmt.Errorf("error or resourceVersion not found: %w", err)
+		}
+
+		if err = unstructured.SetNestedField(req.Object, version, "metadata", "resourceVersion"); err != nil {
+			return fmt.Errorf("couldn't update ResourceVersion: %w", err)
+		}
+
+	}
+	if kind == "Service" {
+		clusterIP, fnd, err := unstructured.NestedString(found.Object, "spec", "clusterIP")
+		if err != nil || !fnd {
+			return fmt.Errorf("error or clusterIP not found: %w", err)
+		}
+
+		if err = unstructured.SetNestedField(req.Object, clusterIP, "spec", "clusterIP"); err != nil {
+			return fmt.Errorf("couldn't update clusterIP: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func SetNodeSelectorTerms(obj *unstructured.Unstructured, terms map[string]string) error {
+
+	if strings.Compare(obj.GetKind(), "DaemonSet") == 0 ||
+		strings.Compare(obj.GetKind(), "Deployment") == 0 ||
+		strings.Compare(obj.GetKind(), "Statefulset") == 0 { // TODO(qbarrand) should this be StatefulSet?
+		if err := nodeSelectorTerms(terms, obj, "spec", "template", "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup %s nodeSelector: %w", obj.GetKind(), err)
+		}
+	}
+	if strings.Compare(obj.GetKind(), "Pod") == 0 {
+		if err := nodeSelectorTerms(terms, obj, "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup Pod nodeSelector: %w", err)
+		}
+	}
+	if strings.Compare(obj.GetKind(), "BuildConfig") == 0 {
+		if err := nodeSelectorTerms(terms, obj, "spec", "nodeSelector"); err != nil {
+			return fmt.Errorf("cannot setup BuildConfig nodeSelector: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func nodeSelectorTerms(terms map[string]string, obj *unstructured.Unstructured, fields ...string) error {
+
+	nodeSelector, found, err := unstructured.NestedMap(obj.Object, fields...)
+	if err != nil {
+		return err
+	}
+
+	if !found {
+		nodeSelector = make(map[string]interface{})
+	}
+
+	for k, v := range terms {
+		nodeSelector[k] = v
+	}
+
+	if err = unstructured.SetNestedMap(obj.Object, nodeSelector, fields...); err != nil {
+		return fmt.Errorf("cannot update nodeSelector for %s : %w", obj.GetName(), err)
+	}
+
+	return nil
+}
+
+func IsOneTimer(obj *unstructured.Unstructured) (bool, error) {
```

Comment:
lets make it a private function

Reply:
We agreed to address this with https://issues.redhat.com/browse/MGMT-8567